src/machine/machine.c

```diff
@@ -523,6 +524,36 @@ int machine_finalize(Machine *m) {
         return 0;
 }
 
+static bool machine_validate_unit(Machine *m) {
+        int r;
+        _cleanup_free_ char *unit = NULL;
+        _cleanup_free_ char *cgroup = NULL;
+
+        r = cg_pid_get_unit(m->leader, &unit);
+        if (!r && streq(m->unit, unit))
```

Comment:
Please don't use boolean comparisons for ints, e.g. `r == 0` instead.

Reply:
done.